hail/src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -60,41 +60,41 @@ final class Region private (empty: Boolean) extends NativeBase() {
   final def allocate(a: Long, n: Long): Long = nativeAlignAllocate(a, n)
   final def allocate(n: Long): Long = nativeAllocate(n)
 
-  private var explicitParents: Int = 0
+  // FIXME: using nativeGetNumParents for now because we're not using `reference` in Scala
```

Comment:
I think I need this bit explained to me, sorry.

Reply:
basically we can use `reference()` to keep a pointer to a region that contains some value that a value in the current region needs to keep around. This basically uses the same mechanism as the whole `setNumParents` thing in c++, but we don't want to mix the two. I tried to keep track of when we were using one and not the other in Scala, but it gets a little messy since we can reuse region objects. I basically gave up on tracking this as I have it written and plan to revisit when we actually start to use `reference` in Scala.